gatling-core/src/main/scala/io/gatling/core/action/TryMax.scala

```diff
@@ -61,11 +59,18 @@ class InnerTryMax(
     case _                           => false
   }
 
-  private def maxNotReached(session: Session): Boolean = session(counterName).validate[Int] match {
-    case Success(i) => i < times
-    case Failure(message) =>
-      logger.error(s"Condition evaluation for tryMax $counterName crashed with message '$message', exiting tryMax")
-      false
+  private def maxNotReached(session: Session): Boolean = {
+    val validationResult = for {
+      counter <- session(counterName).validate[Int]
+      max <- times(session)
+    } yield (counter, max)
+
+    validationResult match {
+      case Success((counter, max)) => counter < max
+      case Failure(message) =>
+        logger.error(s"Condition evaluation for tryMax $counterName crashed with message '$message', exiting tryMax")
+        false
+    }
```

Comment:
Please use one single for-comprehension, without this intermediate `validationResult` and this additional pattern matching.